server/index.js

```diff
@@ -12,17 +20,43 @@ app.use(express.static(path.resolve(__dirname, "../client/build")));
 //Mongoose setup code below
 
 // getting-started.js
-const mongoose = require("mongoose");
 mongoose.connect(
   process.env.HALOGEN_WEB || "mongodb://localhost:3001/showtracker",
   { useNewUrlParser: true, useUnifiedTopology: true }
 );
 const db = mongoose.connection;
 db.on("error", console.error.bind(console, "connection error:"));
 db.once("open", function () {
-  console.log("It's working!");
+  console.log("Mongoose Is Connected");
 });
 
+//----------------------------------------- START OF MIDDLEWARE-----------------------------------------------
+
+app.use(express.json()); //For middleware to parse incoming requests as json
+app.use(express.urlencoded({ extended: true })); //For middleware to recogizes the incoming Request object as strings or arrays
+app.use(
+  cors({
+    //To allow requests to our web server from another domain which in this case is http://localhost:3000
+    origin: "http://localhost:3000", // <-- location of the react app we're connecting to
```

Comment:
I'm assuming we'll have to make some adjustments for this to work on the deployed Heroku app? I haven't tested it, but we can dig in in the future